### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by Rhaegar121**
In this PR, I have completed the following tasks.

1. Created model classes for `User` `Post` `Comment` `Like`
2. Inserted data to models according to the requirements
3. Created custom methods in model classes according to the requirements
4. Fixed all the rubocop errors
5. Tested custom methods in rails console

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post'
+
+  def update_comments_counter
```
Comment (rustam-tolipov) [inline app/models/comment.rb:9]: **You did a great job, I like the implementation you did. Especially seeds.rb file :)**

- [ ] However, I have a suggestion for you to update! For using the update comment counter method we can say that it only belongs to Comment and you do not need to use it in Posts, right? As this comes in handy, why not give it a try to put it after `private` and then call if after_save: update_com... I kindly suggest you do that :)

HINT: [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post'
+
+  def update_likes_counter
```
Comment (rustam-tolipov) [inline app/models/like.rb:9]: - [ ] The same above suggestion works here as well :) Kindly implement the suggestion.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_counter
```
Comment (rustam-tolipov) [inline app/models/post.rb]: - [ ] The last one, this one also works exactly I suggested above :)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Rhaegar121,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Well-documented readme_
_Kudos for adding seeds_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 3 above)
Comment (rustam-tolipov) [reply app/models/post.rb]: Let me know if you have any question :)

**Review by Whoistolu — APPROVED** [review]
### Hi @Rhaegar121 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**